test/tools/jsontests/StateTests.cpp

```diff
@@ -104,10 +105,17 @@ json_spirit::mValue StateTestSuite::doTests(json_spirit::mValue const& _input, b
 						continue;
 					if (test::isDisabledNetwork(test::stringToNetId(i->first)))
 						continue;
-					importer.checkGeneralTestSection(exp.get_obj(), wrongTransactionsIndexes, i->first);
+					if (importer.checkGeneralTestSection(exp.get_obj(), wrongTransactionsIndexes, i->first))
+						foundResults = true;
 				}
 			}
 
+			if (!foundResults)
+			{
+				Options const& opt = Options::get();
+				BOOST_ERROR("Transaction not found! (Network: " + (opt.singleTestNet.empty() ? "Any" : opt.singleTestNet) + ", dataInd: " + toString(opt.trDataIndex) + ", gasInd: " + toString(opt.trGasIndex) + ", valInd: " + toString(opt.trValueIndex) + ")");
```

Comment:
Do we want to cause this error even when `--singletest` is not specified?

Reply:
especially if it is not specified. that would mean that execution logic is broken (no tests executed)

Reply:
Really? `testeth -- -d 3` executes some tests.  Those tests with less than three data are skipped.  Do you want an error for this command?

Reply:
ah, common. it is about an error in a single test suite. 
in yout example above all test suites are executed and some of them would be skipped. 
then I think we should allow -d -v -g --singlenet only if a single test suite (single doTest function)

Reply:
well if this would be a static variable then I could check it after execution all tests. but that would be not elegant.

Reply:
> I think we should allow -d -v -g --singlenet only if a single test suite (single doTest function)

Maybe.  For now, let's error only when `--singletest` is set.

Reply:
hm. so wait. it continues the execution. so you right it might be a warning. just throw a warning if no tests from the test suites are executed.

Reply:
That works for me.

Reply:
hm. boost WARN_MESSAGE  does not print to console.

Reply:
OK.  An error is good then.